src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/RequireEmptyLineBeforeBlockTagsGroupCheck.java

```diff
@@ -0,0 +1,243 @@
+
+import com.puppycrawl.tools.checkstyle.StatelessCheck;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.utils.JavadocUtil;
+
+/**
+ * <p>
+ * Checks that one blank line before the tag block if it is present in Javadoc.
+ * </p>
+ * <ul>
+ * <li>
+ * Property {@code violateExecutionOnNonTightHtml} - Control when to print violations
+ * if the Javadoc being examined by this check violates the tight html rules defined at
+ * <a href="https://checkstyle.org/writingjavadocchecks.html#Tight-HTML_rules">
+ * Tight-HTML Rules</a>.
+ * Type is {@code boolean}. Default value is {@code false}.
+ * </li>
+ * </ul>
+ * <p>
+ * To configure the check:
+ * </p>
+ * <pre>
+ * &lt;module name=&quot;RequireEmptyLineBeforeBlockTagsGroup&quot;/&gt;
+ * </pre>
+ * <p>
+ * By default, the check will report a violation if there is no blank line before the tag block,
+ * like in the example below.
+ * </p>
+ * <pre>
+ * &#47;**
+ *  * testMethod's javadoc.
+ *  * &#64;return something (violation)
+ *  *&#47;
+ * public boolean testMethod() {
+ *     return false;
+ * }
+ * </pre>
+ * <p>
+ *  Valid javadoc should have a blank line separating the parameters, return, throw, or
+ *  other tags like in the example below.
+ *  </p>
+ *  <pre>
+ *  &#47;**
+ *  * testMethod's javadoc.
+ *  *
+ *  * &#64;param firstParam
+ *  * &#64;return something
+ *  *&#47;
+ *  public boolean testMethod(int firstParam) {
+ *      return false;
+ *  }
+ *  </pre>
+ * <p>
+ * Parent is {@code com.puppycrawl.tools.checkstyle.TreeWalker}
+ * </p>
+ * <p>
+ * Violation Message Keys:
+ * </p>
+ * <ul>
+ * <li>
+ * {@code javadoc.missed.html.close}
+ * </li>
+ * <li>
+ * {@code javadoc.parse.rule.error}
+ * </li>
+ * <li>
+ * {@code javadoc.tag.line.before}
+ * </li>
+ * <li>
+ * {@code javadoc.wrong.singleton.html.tag}
+ * </li>
+ * </ul>
+ *
+ * @since 8.36
+ */
+@StatelessCheck
+public class RequireEmptyLineBeforeBlockTagsGroupCheck extends AbstractJavadocCheck {
+
+    /**
+     * The key in "messages.properties" for the message that describes a tag in javadoc
+     * requiring an empty line before it.
+     */
+    public static final String MSG_JAVADOC_TAG_LINE_BEFORE = "javadoc.tag.line.before";
+
+    /**
+     * Case when space separates the tag and the asterisk like in the below example.
+     * <pre>
+     *  /**
+     *   * &#64;param noSpace there is no space here
+     * </pre>
+     */
+    private static final List<Integer> ONLY_TAG_VARIATION_1 = Arrays.asList(
+            JavadocTokenTypes.WS,
+            JavadocTokenTypes.LEADING_ASTERISK,
+            JavadocTokenTypes.NEWLINE);
+
+    /**
+     * Case when no space separates the tag and the asterisk like in the below example.
+     * <pre>
+     *  /**
+     *   *&#64;param noSpace there is no space here
+     * </pre>
+     */
+    private static final List<Integer> ONLY_TAG_VARIATION_2 = Arrays.asList(
```

Comment:
If we remove variation 1, won't 2 always match? If we have to match the entire line or up to the asterisk, isn't it fast to check a size of 2 and the last 2 items are the ones we want?